### PR TITLE
Assignments Schema and changes from upstream/master

### PR DESCRIPTION
**Pull request by parthvshah**
Added fields directly to User for the courses. Assignment and Course created as separate schema.

### CODE REVIEW

**Comment by aniketnk** [comment]
Don't put PR for merging upstream. It gets very messy.

File: server/routes/api/accountManagement.js [hunk 1]
```diff
@@ -1,253 +1,172 @@
 const User = require('../../models/User');
-const UserSession = require('../../models/UserSession');
-const mongoose = require('mongoose');
+const UserSession = require('../../models/UserSession')
+const jwt = require('jsonwebtoken');
+var verifyUser = require('../middleware/Token').verifyUser;
+const fs = require('fs');
+var privateKey = fs.readFileSync('server/sslcert/server.key');
```
[the remark this answers is not in the document]
Comment (parthvshah) [reply server/routes/api/accountManagement.js]: Yes. If I remember correctly.

File: server/models/assignments/Professor.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const mongoose = require('mongoose');
```
[the remark this answers is not in the document]
Comment (parthvshah) [reply server/models/assignments/Professor.js]: Yes. This is just a redundant file. Let it exist.

**Comment by parthvshah** [comment]
I wouldn't have to merge upstream/master if you did it before I committed.

File: server/models/assignments/Course.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+const CourseSchema = new mongoose.Schema({
+  professors: [{
+    type: Schema.Types.ObjectId,
+    ref: 'User'
+  }],
+  students: [{
+    type: Schema.Types.ObjectId,
+    ref: 'User'
+  }],
+  name: {
```
[the remark this answers is not in the document]
Comment (parthvshah) [reply server/models/assignments/Course.js]: Nope, I think this will do fine. With the endpoints and what we're trying to achieve right now.

**Comment by parthvshah** [comment]
Sending a separate PR.

**Pull request closed without merging**